qiskit/algorithms/time_evolvers/variational/solvers/ode/abstract_ode_function.py

```diff
@@ -0,0 +1,51 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Abstract class for generating ODE functions."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Iterable
+
+from qiskit.circuit import Parameter
+
+from ..var_qte_linear_solver import VarQTELinearSolver
+
+
+class AbstractOdeFunction(ABC):
+    """Abstract class for generating ODE functions."""
+
+    def __init__(
+        self,
+        varqte_linear_solver: VarQTELinearSolver,
+        param_dict: dict[Parameter, float],
```

Comment:
For inputs we have been trying to be more flexible in terms of types so I am wondering about `Mapping` rather than `dict` - do we really need the specific type? I see this also in the solve method of var_qte_linear_solver where I guess this gets used - maybe there are other places too that would need changing were this changed.